### ISSUE TITLE
false-positive error inside jest tests for async expect

### ISSUE THREAD

**Issue by lifeart** [playwright-community/eslint-plugin-playwright#79]
Code sample:

```ts
describe('foo', () => {
  it('bar', async () => {
    expect(true).toHaveClass('foo-bar');
  });
});
```

Error sample:

<img width="767" alt="image" src="https://user-images.githubusercontent.com/1360552/185202525-32f0f0fa-432c-4371-9c85-1258e179a657.png">

It seems we have to resolve import name for `expect`, and check it belongs to `playwright` namespace

**Comment by mskelton** [comment]
This has been answered in other issues as well, but basically our recommendation is to use ESLint `overrides` to only use the Playwright/Jest plugins for the appropriate test files by directory, regex pattern, etc.

- https://github.com/playwright-community/eslint-plugin-playwright/issues/66
- https://github.com/playwright-community/eslint-plugin-playwright/issues/54

**Comment by lifeart** [comment]
@mskelton it's quite easy to fix if needed here is sample: https://astexplorer.net/#/gist/c4ca11aa828169071031575a469bcc83/7e3c3446dfd6c612fe7a2bc189691e406caba682

**Comment by mskelton** [comment]
The problem is what if you import expect from another file, like a fixture file?

```js
import {test,expect} from '../my-fixtures'
```

which is perfectly valid and common.

The correct solution is to have the plugin only apply to the files you want it to, such as your Playwright directory.  Testing import names is flaky and prone to false positives.

**Comment by lifeart** [comment]
@mskelton we could completely skip check if `expect` is not imported, it's  jest usecase (expect is global)

**Comment by mskelton** [comment]
Jest has the ability to import those globals:

https://jestjs.io/blog/2020/05/05/jest-26#a-new-way-to-consume-jest---jestglobals

The behavior of this library is fine and encourages good practice of scoping your ESLint plugins to what they should cover.  I don't see this behavior changing.
